services:   
  python-pronostic:
    restart: no
    image: python:alpine3.21
    container_name: python-pronostic
    hostname: python-pronostic
    labels:
      traefik.enable: true
      traefik.http.routers.python-pronostic-liste.tls: true
      traefik.http.routers.python-pronostic-liste.entrypoints: websecure
      traefik.http.routers.python-pronostic-liste.rule: Host(`pronostic.${PERSONAL_DOMAINE_NAME}`) && PathPrefix(`/liste`)
      traefik.http.routers.python-pronostic-liste.middlewares: authelia@docker
      traefik.http.routers.python-pronostic-liste.service: python-pronostic
      traefik.http.routers.python-pronostic.tls: true
      traefik.http.routers.python-pronostic.entrypoints: websecure
      traefik.http.routers.python-pronostic.rule: Host(`pronostic.${PERSONAL_DOMAINE_NAME}`) && !PathPrefix(`/liste`)
      traefik.http.routers.python-pronostic.middlewares: authelia@docker
      traefik.http.routers.python-pronostic.service: python-pronostic
      traefik.http.services.python-pronostic.loadBalancer.server.port: 5000
    environment:
      PUID: 1000
      PGID: 100
      TZ: Europe/Paris
    volumes:
      - /${DOCKER_DATA_DIRECTORY:?data path is required}/repos/home-docker/timezone:/etc/timezone:ro
      - python-pronostic:/data
    env_file:
      - stack.env
    command:
      - sh
      - -c
      - > 
        cd /data &&
        pip install -r requirements.txt && python3 /data/app.py
    networks:
    - webinternal

volumes:
  python-pronostic:
    name: python-pronostic
networks:
  webinternal:
    name: webinternal